### PR TITLE
Dir jo

### PR DESCRIPTION
**Pull request by dmail**
# What does this PR do?

This pull requests improve jsenv dev server UI when listing a directory.
It is heavily inspired from https://github.com/johansatge/httpdir/

# Changes

When the root url is requested and there is no index.html file, jsenv dev server display directory content. It now look like this:

2_dir_many_file

Whenever a file is not found jsenv dev server display the parent directory content. The UI now look as follow:

1_root_many_file

# How to review ?

- The file changes contains files in dist/**, they can be ignored
- The files changes contains some changes in tests/**, they can be ignored
- This PR fixes a small bug in `@jsenv/node-esm-resolution` (applying node resolution on a specifier with a search param would duplicate that search param)

This can be tested as follow:

```console
npx @jsenv/cli@latest
```

It should start the dev server + open a browser. Visit some urls leading to directories and see the new UI.

### CODE REVIEW

File: src/build/build.js [hunk 1]
```diff
@@ -470,9 +470,23 @@ build ${entryPointKeys.length} entry points`);
             addToBundlerIfAny(rawUrlInfo);
           }
           if (rawUrlInfo.type === "html") {
-            rawUrlInfo.referenceToOthersSet.forEach((referenceToOther) => {
+            for (const referenceToOther of rawUrlInfo.referenceToOthersSet) {
```
Comment (dmail) [inline src/build/build.js:473]: Just cosmetic (usage of for-of over forEach)

File: src/build/build_specifier_manager.js [hunk 2]
```diff
@@ -87,9 +87,14 @@ export const createBuildSpecifierManager = ({
 
     let buildSpecifier;
     if (base === "./") {
-      const parentBuildUrl = reference.ownerUrlInfo.isRoot
+      const { ownerUrlInfo } = reference;
+      const parentBuildUrl = ownerUrlInfo.isRoot
         ? buildDirectoryUrl
-        : urlInfoToBuildUrlMap.get(reference.ownerUrlInfo);
+        : urlInfoToBuildUrlMap.get(
```
Comment (dmail) [inline src/build/build_specifier_manager.js:93]: Fix an error when CSS becomes inline during build

File: src/dev/start_dev_server.js [hunk 3]
```diff
@@ -261,6 +261,7 @@ export const startDevServer = async ({
             nodeEsmResolution,
             magicExtensions,
             magicDirectoryIndex,
+            directoryReferenceEffect: "preserve",
```
Comment (dmail) [inline src/dev/start_dev_server.js:264]: Allow dev server to display directory content, before it would have thrown an internal 500 error.

**Review by johansatge — COMMENTED** [review]
Nice ✨ 😃

**Pull request merged**